wasm/func_validation.go

```diff
@@ -0,0 +1,948 @@
+package wasm
```

Comment:
this is extracted from store.go as this can be decoupled from store and a bit complex
compared to others in store.go!